packages/grafana-data/src/datetime/datemath.ts

```diff
@@ -163,15 +163,11 @@ export function parseDateMath(
       return undefined;
     } else {
       if (type === 0) {
-        if (roundUp) {
-          if (isFiscal) {
-            roundToFiscal(fiscalYearStartMonth, dateTime, unit, roundUp);
-          } else {
-            dateTime.endOf(unit);
-          }
+        if (isFiscal) {
+          roundToFiscal(fiscalYearStartMonth, dateTime, unit, roundUp);
         } else {
-          if (isFiscal) {
-            roundToFiscal(fiscalYearStartMonth, dateTime, unit, roundUp);
+          if (roundUp) {
+            dateTime.endOf(unit);
           } else {
             dateTime.startOf(unit);
           }
```

Comment:
this is a small bit of refactoring which isn't actually relevant to the fix

Reply:
It does look clearer though 👏